packages/react-storage/src/components/StorageBrowser/providers/store/location/context.tsx

```diff
@@ -0,0 +1,107 @@
+import React from 'react';
+
+import { createContextUtilities } from '@aws-amplify/ui-react-core';
+import { noop } from '@aws-amplify/ui';
+
+import { assertLocationData } from '../../../validators';
+import { LocationData } from '../../../actions';
+
+export const ERROR_MESSAGE =
+  'Invalid `location` value provided as initial value to `LocationProvider.';
+
+export const DEFAULT_STATE: LocationState = {
+  current: undefined,
+  path: undefined,
+};
+
+export type LocationActionType = {
+  type: 'NAVIGATE';
+  location?: LocationData;
+  path?: string;
```

Comment:
As part of the navigate action type, it kind of just unnecessarily requires an explicit `path: undefined` when just omitting it would do. I think making the state provided path default to `''` will be more helpful to consumers of the state